tests/test_retries.py

```diff
@@ -237,3 +237,43 @@ def test_retry_respect_retry_after_headers():
     all_verbs = [v for v in (verbs + [v.lower() for v in verbs])]
     with cf.ThreadPoolExecutor() as executor:
         executor.map(_retry_respect_retry_after_headers, all_verbs)
+
+
+@pytest.mark.parametrize(
+    "retrying, max_calls",
+    [
+        # Simulate a user-provided tenacity.Retrying with limited max attempts.
+        (_get_retrying(3), 3),
+        # No user-provided tenacity.Retrying. Use civis-python's default retrying.
+        (None, 10),
+    ],
+)
+@mock.patch("civis.futures.time.sleep", side_effect=lambda x: None)
+def test_no_recursion_in_retry_request(m_sleep, retrying, max_calls):
```

Comment:
New unit test based on #524. I've modified the test code to test for both cases of a user-specified retryer and the default retryer.